CHANGELOG.adoc

```diff
@@ -85,6 +85,37 @@ Additionally, `ErrorStatusCode` provides an `as_client_error` method that
 returns a `ClientErrorStatusCode` if the status code is a client error, or an
 error.
 
+* `HttpError` now contains an `Option<Box<http::HeaderMap>>` of headers to add
+to error responses constructed for the error.
++
+Code which constructs `HttpError` literals must now initialize this field. For
+example, this will no longer compile:
++
+```rust
+let err = dropshot::HttpError {
+    status_code: ErrorStatusCode::INTERNAL_SERVER_ERROR,
+    error_code: None
+    internal_message: "something bad happened".to_string(),
+    external_message: "Internal Error".to_string(),
+};
+```
++
+Instead, write:
++
+```rust
+let err = dropshot::HttpError {
+    status_code: ErrorStatusCode::INTERNAL_SERVER_ERROR,
+    error_code: None
+    internal_message: "something bad happened".to_string(),
+    external_message: "Internal Error".to_string(),
+    headers: None, // <-- Add this
+};
+```
++
+Similarly, code which destructures `HttpError`s, such as in a `match` or `let`
+expression, must now either destructure the `headers` field or add `..` to
+ignore it.
```

Comment:
del

Reply:
IMO, this is worth mentioning, because otherwise, the language in the previous sentence suggests that this change _only_ applies to code that constructs a `HttpError` literal. We could simplify it to something like:

> Code which constructs a `HttpError` literal or destructures `HttpError`s in a pattern must be changed to add this field.

What do you think?

Reply:
All this feels like the expected consequences of adding a new field to a structure. I don't expect folks will proactively fix things based on the changelog and I also don't think folks will go looking at the changelog to figure out how to address these compilation errors.